torch/csrc/distributed/c10d/Types.hpp

```diff
@@ -137,6 +137,7 @@ struct ReduceOptions {
 
 struct AllgatherOptions {
   std::chrono::milliseconds timeout = kUnsetTimeout;
+  bool asyncOp = true;
```

Comment:
should these be `true` by default? Our current collectives have` async = false` by default right?

Reply:
For the python APIs, yes, `async = false` is the default.
But for C++ APIs, i.e. APIs defined by the `Backend` class, `async = true` is the default behavior.

Reply:
Here `struct AllgatherOptions` is mainly for passing the async option **into** the C++ implementations, so I chose `true` as the default behavior.